pkg/descheduler/strategies/duplicates.go

```diff
@@ -75,18 +89,21 @@ func listDuplicatePodsOnANode(ctx context.Context, client clientset.Interface, n
 				podKeys = append(podKeys, s)
 			}
 		}
+		sort.Strings(podKeys)
```

Comment:
Assuming the following pods owned by the same (namespace, kind, owner) triple:
```
- pod1:
  - image1
  - image2
- pod2:
  - image2
  - image1
```
Do we want to consider both pods to be duplicates of each other?

Also, the following case independent of ordering where one image contains multiple binaries:
```
- pod1:
  - image1 (containername=withbinary1)
  - image1 (containername=withbinary2)
- pod2:
  - image1 (containername=withbinary3)
  - image1 (containername=withbinary4)
```
Each binary responsible for a different aspect of a customer solution. We could keep going and discuss the case where two pods have the exact same list of containers (wrt. image and container name), yet both pods running different binaries. Which will make the logic unnecessarily complicated. 

This can eventually lead to customers reporting strange behavior for their custom controllers. As long as the owner kind is a known controller (RC, RS, SS, Deployment, ..), we are safe. However, we need to stress in the strategy description it's meant to be run only over template based owners. Or, allow to explicitly skip specific list of kinds. @damemi what do you think about adding a new `ignoreOwnerKinds` param to the strategy?

Reply:
We talked about this a little offline so for reference here, I think that those 2 cases are reasonably out of the scope for what we should take into consideration for a duplicate pod.

The ordering of containers in a pod doesn't have any effect on their actual output, so I think it's safe to ignore that. And we can't really know what binaries are in the image, but we could look at the commands that are being run in the container. I think that's excessive though, but if there ends up being demand for that level of specificity I'm open to revisiting it.

Ultimately with these changes we have already made the strategy much stricter than it was before, which is a big improvement. I also pushed commit https://github.com/kubernetes-sigs/descheduler/pull/275/commits/c11cb4f52bea5c71e3fa519b6a58aabdcdce13a4, which adds a parameter to exclude certain kinds of ownerrefs as you suggested (and also addresses the request in https://github.com/kubernetes-sigs/descheduler/issues/243). 

One caveat to note about this parameter is that it becomes all-or-nothing for considering a pod a duplicate. That is, if the pod has multiple ownerrefs and one of them is an excluded kind, we have to consider the whole pod ineligible for eviction. Because for example, if we just skipped adding the keys for that particular ownerref, it would actually make the pod *more* likely to match another pod (as it would have less keys identifying its uniqueness)

Please let me know if the new parameter matches what you had in mind, and if so I can squash the commits down.

Reply:
Looks good!!! Please squash, I will lgtm then.